src/app/zap-templates/call-command-handler-src.zapt

```diff
@@ -100,15 +100,64 @@ EmberAfStatus emberAf{{asCamelCased name false}}Cluster{{asCamelCased side false
         {{#if (isCommandAvailable parent.side incoming outgoing commandSource name)}}
         case ZCL_{{asDelimitedMacro name}}_COMMAND_ID: {
             {{#if (zcl_command_arguments_count this.id)}}
-            uint32_t argOffset = 0;
+            uint32_t payloadOffset = cmd->payloadStartIndex;
             {{#zcl_command_arguments}}
-            {{asUnderlyingType type}} * {{asSymbol label}} = ({{asUnderlyingType type}} *)(cmd->buffer + argOffset);
+            {{#if (isNullable type isArray)}}
+            uint8_t * {{asSymbol label}};
+            {{else if (isStrEqual label "groupId")}}
+            GroupId {{asSymbol label}};
+            {{else}}
+            {{asUnderlyingType type}} {{asSymbol label}};
+            {{/if}}
+            {{/zcl_command_arguments}}
+
+            {{#zcl_command_arguments}}
+            {{#if presentIf}}
+            if ({{presentIf}})
+            {
+            {{/if}}
+            {{#if isArray}}
+            {{asSymbol label}} = (uint8_t *)(cmd->buffer + payloadOffset);
+            {{else if (isString type)}}
+            {{asSymbol label}} = emberAfGetString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isLongString type)}}
+            {{asSymbol label}} = emberAfGetLongString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isStrEqual label "groupId")}}
+            {{asSymbol label}} = (*(GroupId *)(cmd->buffer + payloadOffset));
```

Comment:
This doesn't look right.  Should be `emberAfGetInt16u(cmd->buffer, payloadOffset, cmd->bufLen);`, I would think, for now.